src/newrelic/collector/collector.go

```diff
@@ -26,7 +26,7 @@ const (
 )
 
 const (
-	protocolVersion = "16"
+	protocolVersion = "17"
```

Comment:
Does the protocol version update belong to this PR? Maybe it does once it is re-based onto the code that addresses the P17 response code handling, i.e. after #418 is closed.

Reply:
I see your point - by the book yes we probably would rather have the flag flipped in that PR.  Realistically since all dev is currently on P17 and the agent does appear to work for the purposes of P17 dev as is I think this will be OK since the #418 PR will not be too far away.  But I am fine if the consensus is to hold off on flipping this now.

Reply:
On the second thought, the earlier we have the updated the protocol version, the earlier other parts that depend on P17 being enabled can be tested (thinking labels, logs in context). Since it works, I'm ok with including protocol version update in this PR.

Reply:
That was my reasoning - I probably should have run it by everyone first.  :)